src/python/pants/option/global_options.py

```diff
@@ -715,7 +771,46 @@ def register_bootstrap_options(cls, register):
             # This default is also hard-coded into the engine's rust code in
             # fs::Store::default_path so that tools using a Store outside of pants
             # are likely to be able to use the same storage location.
-            default=os.path.join(get_pants_cachedir(), "lmdb_store"),
+            default=DEFAULT_LOCAL_STORE_OPTIONS.store_dir,
+        )
+        register(
+            "--local-store-processes-max-size-bytes",
+            type=int,
+            advanced=True,
+            help=(
+                "The maximum size in bytes of the local store containing process cache entries. "
+                f"Stored below `{local_store_dir_flag}`."
+            ),
+            default=DEFAULT_LOCAL_STORE_OPTIONS.processes_max_size_bytes,
+        )
+        register(
+            "--local-store-files-max-size-bytes",
+            type=int,
+            advanced=True,
+            help=(
+                "The maximum size in bytes of the local store containing files. "
+                f"Stored below `{local_store_dir_flag}`."
+                "\n\n"
+                "NB: This size value bounds the total size of all files, but (due to sharding of the "
+                "store on disk) it also bounds the per-file size to (VALUE / 16)."
+                "\n\n"
+                "This value doesn't reflect space allocated on disk, or RAM allocated (it "
+                "may be reflected in VIRT but not RSS). However, the default is lower than you "
+                "might otherwise choose because macOS creates core dumps that include MMAP'd "
+                "pages, and setting this too high might cause core dumps to use an unreasonable "
+                "amount of disk if they are enabled."
+            ),
+            default=DEFAULT_LOCAL_STORE_OPTIONS.files_max_size_bytes,
+        )
+        register(
+            "--local-store-directories-max-size-bytes",
+            type=int,
+            advanced=True,
+            help=(
+                "The maximum size in bytes of the local store containing directories. "
+                f"Stored below `{local_store_dir_flag}`."
+            ),
+            default=DEFAULT_LOCAL_STORE_OPTIONS.directories_max_size_bytes,
         )
```

Comment:
One potential alternative to exposing all three options would be to expose one option, and to then bake in a ratio.